tests/manage/z_cluster/nodes/test_disk_failures.py

```diff
@@ -169,87 +171,165 @@ def test_recovery_from_volume_deletion(self, nodes, pvc_factory, pod_factory):
+        osd_id = osd_pod.get().get('metadata').get('labels').get('ceph-osd-id')
 
         # Get the node that has the OSD pod running on
-        logger.info(f"Getting the node that has the OSD pod {osd_pod.name} running on")
+        logger.info(
+            f"Getting the node that has the OSD pod {osd_pod.name} running on"
+        )
         osd_node = get_pod_node(osd_pod)
-        volume_size = osd_pvc.size
         osd_prepare_pods = get_osd_prepare_pods()
         osd_prepare_pod = [
             pod for pod in osd_prepare_pods if pod.get().get('metadata')
             .get('labels').get(constants.CEPH_ROOK_IO_PVC_LABEL) == claim_name
         ][0]
-        osd_prepare_job_name = osd_prepare_pod.get().get('metadata').get('labels').get('job-name')
+        osd_prepare_job_name = osd_prepare_pod.get().get(
+            'metadata').get('labels').get('job-name')
         osd_prepare_job = get_job_obj(osd_prepare_job_name)
 
         # Get the corresponding OSD deployment
-        logger.info(f"Getting the corresponding OSD deployment for OSD PVC {claim_name}")
+        logger.info(f"Getting the OSD deployment for OSD PVC {claim_name}")
         osd_deployment = [
             osd_pod for osd_pod in get_osd_deployments() if osd_pod.get()
-            .get('metadata').get('labels').get(constants.CEPH_ROOK_IO_PVC_LABEL) == claim_name
+            .get('metadata').get('labels')
+            .get(constants.CEPH_ROOK_IO_PVC_LABEL) == claim_name
         ][0]
+        osd_deployment_name = osd_deployment.name
 
         # Delete the volume from the platform side
-        logger.info(f"Deleting volume {backing_volume} from the platform side")
+        logger.info(f"Deleting {backing_volume} from the platform side")
         nodes.detach_volume(backing_volume, osd_node)
 
-        # Delete the OSD deployment
-        osd_deployment_name = osd_deployment.name
-        logger.info(f"Deleting OSD deployment {osd_deployment_name}")
-        osd_deployment.delete()
-        osd_deployment.ocp.wait_for_delete(resource_name=osd_deployment_name, timeout=120)
+        # Scale down OSD deployment
+        logger.info(f"Scaling down OSD deployment {osd_deployment_name} to 0")
+        ocp.OCP().exec_oc_cmd(
+            f"scale --replicas=0 deployment/{osd_deployment_name}"
+        )
+
+        # Force delete OSD pod if necessary
+        osd_pod_name = osd_pod.name
+        logger.info(f"Waiting for OSD pod {osd_pod.name} to get deleted")
+        if osd_pod.ocp.wait_for_resource(
+            condition=constants.STATUS_TERMINATING, resource_name=osd_pod_name
+        ):
+            osd_pod.delete(force=True)
+        osd_pod.ocp.wait_for_delete(resource_name=osd_pod_name)
+
+        # Run ocs-osd-removal job
+        logger.info(f"Executing OSD removal job on OSD-{osd_id}")
+        osd_removal_job_yaml = ocp.OCP().exec_oc_cmd(
+            f"process -n openshift-storage ocs-osd-removal"
+            f" -p FAILED_OSD_ID={osd_id} -o yaml"
+        )
```

Comment:
exec_oc_cmd will append -n namespace.
```suggestion
        osd_removal_job_yaml = ocp.OCP(
        namespace=config.ENV_DATA['cluster_namespace']
        ).exec_oc_cmd(
            f"process ocs-osd-removal"
            f" -p FAILED_OSD_ID={osd_id} -o yaml"
        )
```

Reply:
Ack. Made the required change

Reply:
Missed to add namespace parameter ?
```
ocp.OCP(
        namespace=config.ENV_DATA['cluster_namespace']
        )
```